cmd/snap/cmd_run.go

```diff
@@ -1186,7 +1186,13 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	// https://www.freedesktop.org/wiki/Software/systemd/ControlGroupInterface/
 	_, appName := snap.SplitSnapApp(snapApp)
 	needsTracking := true
-	if app := info.Apps[appName]; hook == "" && app != nil && app.IsService() {
+
+	// Do not track apps that don't wish to be tracked.
+	if app := info.Apps[appName]; hook == "" && app.Untracked {
+		needsTracking = false
+	}
+
+	if app := info.Apps[appName]; needsTracking && hook == "" && app != nil && app.IsService() {
```

Comment:
nitpick: I'd prefer an `else` over adding `needsTracking` to a conditional which is already on the long side.
```suggestion
	app := info.Apps[appName]
	if hook == "" && app.Untracked {
		// Do not track apps that don't wish to be tracked.
		needsTracking = false
	} else if hook == "" && app != nil && app.IsService() {
```
Not that I like this a lot either, but IMHO it's a bit easier to follow.

Reply:
This part of the diff is gone now.